### ISSUE TITLE
MvxViewModel<TParameter> occurs exception.

### ISSUE THREAD

**Issue by jz5** [MvvmCross/MvvmCross#2028]
### Steps to reproduce

1. Install MvvmCross.Plugins.Json
2. Call Navigate<TViewModel, TParameter> in view model

```cs
await Mvx.Resolve<IMvxNavigationService>().Navigate<SecondViewModel, Parameter>(
    new Parameter
    {
        Foo = 100,
        Bar = "Baz",
    }
);
```
3. MvxViewModel<TParameter> class occurs exception. (HockeyApp reports this exception)

### Expected behavior

No exception.
I do not know the solution. Is ``Task Init(string parameter)`` need?

### Actual behavior

``public async Task Init(string parameter)`` method called with null parameter(string parameter is null). So ```var deserialized = serializer.DeserializeObject<TParameter>(parameter);``` occurs exception. 

```cs
public abstract class MvxViewModel<TParameter> : MvxViewModel, IMvxViewModel<TParameter> where TParameter : class
{
    public async Task Init(string parameter) // *** It called with null parameter when Navigate called ***
    {
        IMvxJsonConverter serializer;
        if (!Mvx.TryResolve(out serializer)) // *** It is true when MvvmCross.Plugins.Json is installed ***
        {
            throw new MvxIoCResolveException("There is no implementation of IMvxJsonConverter registered. You need to use the MvvmCross Json plugin or create your own implementation of IMvxJsonConverter.");
        }

        var deserialized = serializer.DeserializeObject<TParameter>(parameter); // *** exception occurred when parameter is null ***
        await Initialize(deserialized);
    }

    public abstract Task Initialize(TParameter parameter);
}
```

App doesn't crush.  But HockeyApp reports this inner exception.

### Configuration

**Version:** 5.0.5
And MvvmCross.Plugin

**Platform:** 
- [ ] iOS
- [x] Android
- [ ] WPF
- [ ] UWP

**Comment by jz5** [comment]
Minimum project for reproduce: https://github.com/jz5/MvxApp2028

**Comment by nmilcoff** [comment]
Thanks for reporting this! It should be fixed by #2041
